src/runtime/exceptions.cs

```diff
@@ -76,12 +77,27 @@ public new static IntPtr tp_str(IntPtr ob)
             {
                 message = e.Message;
             }
+            if (e is AggregateException ae)
+            {
+                message += GetAggregateExceptionString(ae);
+            }
```

Comment:
It is necessary because most users will not want or care to unpack an AggregateException.  They just want a meaningful error message by default.

Reply:
What I mean is that this should not be done at `SetError` level. The caller of `SetError` can handle it on their own. This violates single responsibility.

Reply:
The underlying problem is that this method doesn't call ToString on the exception.  It assumes that the exception message has all relevant information, which is not true for AggregateException.  I can change it to call ToString instead.

Reply:
BTW ClassBase.tp_str does call ToString.  For some reason, ExceptionClassObject changes this behavior for exceptions.

Reply:
Actually, that makes sense. It should just call `ToString`. I think custom `tp_str` was made to avoid "System.Exception: " prefix in the message.

Perhaps we could strip the prefix if the result of `ToString()` starts with it.

Reply:
For all exceptions, or just AggregateException?

Reply:
For all.